### PR TITLE
Fix CommandReader hanging at shutdown on Windows

### PR DESCRIPTION
**Pull request by dktapps**
## Description

Windows doesn't do non-blocking streams, which is somewhat irritating. For the most part this won't affect a thing, because it just means the CommandReader thread is blocked until a command is entered. However, on shutdown this causes issues, because the CR needs input to escape that blocking fgets() causing it to hang. Previously this was noticeable because the thread would continue to hang until pressing Enter. (More recently, this hasn't been noticeable due to a pthreads bug causing a shutdown crash.)

This PR fixes the issue by using stream_select to check for changes in stdin *before* making the blocking fgets() call. This brings the behaviour more in line with Linux. This is not so much a fix as it is a workaround for a very old issue.

## Tests
Tested on Android and Windows, but not thoroughly. I noticed no issues under any circumstances.
This modification should not affect any platform except Windows.

### CODE REVIEW

**Comment by ishitatsuyuki** [comment]
I think the current impl is quite wasting the CPU.
Before: 3.x% waste
Genisys one: <0.1% waste (if not infinite-looping by bug)

**Comment by dktapps** [comment]
@ishitatsuyuki I opted for the absolute minimum required to fix the bug. I can work on performance improvements later. Also, Genisys CR _never_ shuts down correctly. Heap corruption or segfault, every time.

**Comment by ishitatsuyuki** [comment]
Neither does PM. This is a little bit inefficient, laggy and bad.

**Comment by dktapps** [comment]
@ishitatsuyuki I have never had issues with PM's CR, except on Windows, because of the blocking issue I described above, which is corrected by simply hitting Enter

**Comment by SOF3** [comment]
:+1: for no hardcoding if(os == Windows)

File: src/pocketmine/command/CommandReader.php [hunk 1]
```diff
@@ -77,7 +87,7 @@ public function run(){
 		}else{
 			global $stdin;
 			$stdin = fopen("php://stdin", "r");
-			stream_set_blocking($stdin, 0);
+			$this->blocking = !(stream_set_blocking($stdin, 0)); //This always fails on Windows
```
Comment (SOF3) [inline src/pocketmine/command/CommandReader.php]: Unnecessary parentheses

**Comment by ishitatsuyuki** [comment]
It's up to you, though. Since this is a PHP implementation issue, workarounds should be specific to a platform.

**Comment by dktapps** [comment]
@ishitatsuyuki This fix won't change anything on non-Windows platforms. This will only affect platforms which suffer the blocking issue (i.e. Windows). https://github.com/iTXTech/PocketMine-MP/pull/4/files#diff-2e903f62b4f3195f5d71c6f3345e5e24R90

File: src/pocketmine/command/CommandReader.php (hunk 1 above)
Comment (dktapps) [inline src/pocketmine/command/CommandReader.php]: :+1:

**Comment by dktapps** [comment]
Hmm, ran into some issues. Holding this for now.

**Pull request closed without merging**